src/entry_points/markdownlintcli2_soft_fail.py

```diff
@@ -0,0 +1,34 @@
+"""
+Lint the markdown in ./README.md and ./docs/ using the external command
+markdownlint-cli2.
+
+This script's purpose is:
+
+1. Run the external command markdownlint-cli2 as a subprocess if it's
+    installed, if not fail silently. The aim is to avoid disruption to
+    external contributors who may not even be working on the docs.
+2. Keep tox cross-platform by not insisting on any external commands
+    (including bash).
+"""
+
+import shutil
+import subprocess
+import sys
+
+
+def main():
+    """
+    Run markdownlint-cli2 as a subprocess if it's installed, if not fail silently.
+    """
+    markdownlint = shutil.which("markdownlint-cli2")
+    if not markdownlint:
+        # Note: There's an additional step in test.yaml to run markdownlint-cli2 in github actions
+        print("*********  Install 'markdownlint-cli2' to enable markdown linting *********")
+        sys.exit(0)
```

Comment:
Sorry if I missed something obvious here but, don't we need `GITHUB_ACTIONS` check to fail here too?

Reply:
Ooh, this is a little inconsistent, Markdown lint is handled separately by the [markdownlint-cli2-action](https://github.com/marketplace/actions/markdownlint-cli2-action) in Github Actions:
https://github.com/ethereum/execution-spec-tests/blob/a77f3939df6f676abf0a09d96a3dc6e6aadbbc36/.github/workflows/test.yaml#L49